src/kontrol/kdist/cheatcodes.md

```diff
@@ -907,6 +913,53 @@ Otherwise, throw an error for any other call to the Foundry contract.
       [owise]
 ```
 
+```k
+    rule [foundry.call.mockCall]:
+         <k> #call_foundry SELECTOR ARGS
+          => #loadAccount #asWord(#range(ARGS, 0, 32)) 
+          ~> #setMockCall #asWord(#range(ARGS, 0, 32)) #range(ARGS, #asWord(#range(ARGS, 32, 32)) +Int 32, #asWord(#range(ARGS, #asWord(#range(ARGS, 32, 32)), 32))) #range(ARGS, #asWord(#range(ARGS, 64, 32)) +Int 32, #asWord(#range(ARGS, #asWord(#range(ARGS, 64, 32)), 32)))
+         ...
+         </k> 
+      requires SELECTOR ==Int selector ( "mockCall(address,bytes,bytes)" )
+
+    //rule [foundry.call.mockCallWithMsgValue]:
+    //     <k> #call_foundry SELECTOR ARGS
+    //        => #loadAccount #asWord(#range(ARGS, 0, 32))
+    //        ~> #setMockCall #asWord(#range(ARGS, 0, 32)) #range(ARGS, 96, #asWord(#range(ARGS, 64, 32))) false ... </k>
+    //  requires SELECTOR ==Int selector ( "mockCall(address,uint256,bytes,bytes)" )
+```
+
+```k
+    rule [foundry.set.mockCall]:
+         <k> #next [ _OP:CallOp ] ~> (. =>  #setLocalMem RETSTART RETWIDTH RETURNDATA) ~> #execute ... </k>
+         <localMem> LM </localMem>
+         <wordStack> _ : ACCTTO : _ : ARGSTART : ARGWIDTH : RETSTART : RETWIDTH : _WS </wordStack>
+         <mockCall>
+           <mockAddress> ACCTTO </mockAddress>
+           <mockValues>...  CALLDATA |-> RETURNDATA ...</mockValues>
+         </mockCall>
+         requires #range(LM, ARGSTART, ARGWIDTH) ==K #range(CALLDATA,0,ARGWIDTH)
+      [priority(30)]
+
+      rule [foundry.set.mockCall2]:
+         <k> #next [ _OP:CallSixOp ] ~> (. =>  #setLocalMem RETSTART RETWIDTH RETURNDATA) ~> #execute ... </k>
+         <localMem> LM </localMem>
+         <wordStack> _ : ACCTTO : ARGSTART : ARGWIDTH : RETSTART : RETWIDTH : _WS </wordStack>
+         <mockCall>
+           <mockAddress> ACCTTO </mockAddress>
+           <mockValues>...  CALLDATA |-> RETURNDATA ...</mockValues>
+         </mockCall>
+         requires #range(LM, ARGSTART, ARGWIDTH) ==K #range(CALLDATA,0,ARGWIDTH)
+      [priority(30)]
+```
+
+```k
+    syntax KItem ::= "#updateOutput" Bytes [klabel("foundry_updateRevertOutput")]
+ // -------------------------------------------------------------------------------------
```

Comment:
```suggestion
 // -----------------------------------------------------------------------------
```

Reply:
Also, where is this function used?

Reply:
It's not being used anymore. I will delete it.